components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediatorFactory.java

```diff
@@ -0,0 +1,202 @@
+/*
+ *
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  you may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.micro.integrator.mediator.dataservice;
+
+import org.apache.axiom.om.OMAttribute;
+import org.apache.axiom.om.OMElement;
+import org.apache.synapse.Mediator;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.config.xml.AbstractMediatorFactory;
+import org.apache.synapse.config.xml.SynapseJsonPathFactory;
+import org.apache.synapse.config.xml.SynapsePath;
+import org.apache.synapse.config.xml.SynapseXPathFactory;
+import org.apache.synapse.config.xml.XMLConfigConstants;
+import org.apache.synapse.util.xpath.SynapseXPath;
+import org.jaxen.JaxenException;
+import org.wso2.micro.integrator.mediator.dataservice.DataServiceCallMediator.OperationsType;
+import org.wso2.micro.integrator.mediator.dataservice.DataServiceCallMediator.Operations;
+import org.wso2.micro.integrator.mediator.dataservice.DataServiceCallMediator.Param;
+
+
+import javax.xml.namespace.QName;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Factory for {@link DataServiceCallMediator} instances.
+ * <p>
+ * Configuration syntax:
+ * <pre>
+ * &lt;dataServicesCall serviceName = ""&gt;
+ *      &lt;operations type="single"/&gt;
+ *          &lt;operation name=""&gt;
+ *                  &lt;param name="string"&gt; &lt;param/&gt;
+ *          &lt;/operation&gt;
+ *      &lt;/operations&gt;
+ *      &lt;target  type="envelope"|"property" name="string"/&gt;
```

Comment:
should this be envelope of body ?

Reply:
Yes we are replacing envelop. But since envelop is more convenient to use in this context, it is set as that. WDYT?

Reply:
ok +1